package.json

```diff
@@ -25,12 +25,53 @@
     "babel-plugin-transform-runtime": "^6.5.0",
     "babel-preset-es2015": "^6.5.0",
     "babel-preset-react": "^6.5.0",
+    "babel-eslint": "^7.0.0",
     "gulp-babel": "^6.0.0",
     "gulp-cached": "^1.0.4",
     "gulp-concat": "^2.5.2",
     "gulp-file": "^0.2.0",
     "gulp-order": "^1.1.1",
     "gulp-remember": "^0.3.0",
+    "gulp-eslint": "^3.0.1",
+    "gulp-if": "^2.0.1",
+    "gulp-tap": "^0.1.3",
+    "eslint-plugin-react": "^6.4.0",
+    "eslint-plugin-import": "^1.16.0",
+    "eslint-plugin-jsx-a11y": "^2.2.3",
+    "eslint-config-airbnb": "^12.0.0",
     "streamqueue": "^0.1.3"
+  },
+  "eslintConfig": {
+    "extends": "eslint-config-airbnb",
+    "rules": {
+      "import/no-unresolved": "off",
+      "import/no-extraneous-dependencies": "off",
+      "import/extensions": ["error", "never"],
+      "react/jsx-filename-extension": ["error", { "extensions": [".js"] }],
+      "react/react-in-jsx-scope": "off",
+      "react/prop-types": "off",
+      "react/sort-comp": ["error", {
+        "order": [
+          "/^constructor$/",
+          "/^init$/",
+          "/^initProps$/",
+          "static-methods",
+          "/^content$/",
+          "/^view$/",
+          "/^config$/",
+          "everything-else",
+          "/^onsubmit$/",
+          "/^onerror$/"
+        ]
+      }]
```

Comment:
The order is the order methods must come in classes. I looked for the most common pattern for methods and came up with this order. As for the rest, the documentation explains it better than I could.